### PR TITLE
Refactored to support latest Firefox

### PR DESCRIPTION
**Pull request by Akkuma**
All deprecated functionality that had been removed was replaced with the most current ways of doing things, aside from functionality deprecated, but not relocated yet.

@nzack @JasonGhent and not sure who else might all want to look at this. This is a giant pull party with floaties provided for those that need them.

### CODE REVIEW

File: lib/extended-page-mod.js [hunk 1]
```diff
@@ -333,92 +341,90 @@ const ExtendedPageMod = Loader.compose(EventEmitter, {
...
+    if (!count(this, 'error')) console.error(e);
   },
-  has: function has(instance) {
+  has(instance) {
     let _registry = this._registry;
     return (
-      (0 <= _registry.indexOf(instance)) ||
-      (instance && instance._public && 0 <= _registry.indexOf(instance._public))
+      (0 <= _registry.indexOf(instance)) /*||
+        (instance && instance._public && 0 <= _registry.indexOf(instance._public))//*/
```
Comment (nzack) [inline lib/extended-page-mod.js]: what was this check doing?  just trying to understand why it was removed.

File: lib/extended-page-mod.js [hunk 2]
```diff
@@ -436,23 +442,29 @@ const ExtendedPageModManager = Registry.resolve({
       return;
     }
 
-    for (let rule in RULES)
-      if (RULES[rule].test(document.URL)) {
-        // If the content script is already injected we want to just exit
-        if (window.__contentScriptInjected) {
-          return;
-        }
+    if (window.__contentScriptInjected) return;
```
Comment (nzack) [inline lib/extended-page-mod.js]: nit.  don't one-line if statements that involve transferring code control. it's easier to miss them

File: lib/extended-page-mod.js [hunk 3]
```diff
@@ -436,23 +442,29 @@ const ExtendedPageModManager = Registry.resolve({
...
+
+    //@TODO:Determine if we need to iterate every instance & test or
+    //is the first one enough
+    this._registry.some(instance => {
+      return instance.include.some((pattern, url) => {
+        if (!pattern.test(document.URL)) return false;
```
Comment (nzack) [inline lib/extended-page-mod.js]: ditto

**Comment by nzack** [comment]
Is there any additional context that can be added into the docs or the README that will help us in the future when Firefox decides to deprecate additional functionality?

**Comment by JasonGhent** [comment]
@ravenac95, only you have access to merge to this repo, AFAIK. Can you take a look?

**Comment by nzack** [comment]
@JasonGhent I have merge access as well

File: lib/extended-page-mod.js (hunk 1 above)
Comment (Akkuma) [inline lib/extended-page-mod.js]: So in the old `compose`/`resolve`, it appeared to be the real instance of the class. You couldn't do `fluffy instanceof Dog`. You'd have to do `fluffy._public instanceofDog`.

**Comment by JasonGhent** [comment]
I tested this locally and it looks good.

**Pull request merged**